rpc-proxy/Msg/DBus.hs

```diff
@@ -97,6 +97,10 @@ incomingData sock =
     ) `E.catch` err
   where
     err :: E.SomeException -> IO LazyBS.ByteString
-    err x = do warn $ printf "%s on %s" (show x) (show sock)
-               return LazyBS.empty
-
+    err x = do
+      case E.fromException x of
+       Just (EOF s) -> do debug $ printf "%s on %s" (show x) (show sock)
+                          return LazyBS.empty
+       Just ex -> do warn $ printf "%s on %s" (show x) (show sock)
+                     return LazyBS.empty
+       Nothing -> do return LazyBS.empty
```

Comment:
Couldn't this be pattern matched? like:
```
err Just (EOF s) = do debug [...]
err Just ex = do warn [...]
err Nothing = do [...]
```
Also, `Just ex` should be replaceable by "Just _", since the value doesn't seem to be used.

Reply:
Can't be pattern matched like you have above because you need to call E.fromException.  ```Just ex``` can certainly be replaced by _.

Reply:
Ack tx, I'll never understand haskell :)

Reply:
These two options compile.   I haven't tested through and don't know if changing `show x` to `show s` or `show ex` change the output.  Are either preferred, or should I just make the `Just _` change?

```
    err :: E.SomeException -> IO LazyBS.ByteString
    err x = do _err $ E.fromException x
                return LazyBS.empty
    _err (Just (EOF s)) = debug $ printf "%s on %s" (show s) (show sock)
    _err (Just ex) = warn $ printf "%s on %s" (show ex) (show sock)
    _err Nothing = return ()
```
or
```
    err :: E.SomeException -> IO BL.ByteString 
    err x = do
      case E.fromException x of
        Just (EOF s) -> do debug $ printf "%s on %s" (show x) (show sock)
        Just ex -> do warn $ printf "%s on %s" (show x) (show sock)
        Nothing -> do return ()
      return BL.empty
```

Reply:
They both look better IMO.
I like the first one because it's a bit more "functional", and the second one because it's cleaner.
Sorry for nit picking on this, honestly the original one is fine too if you don't have time to test another option.

Reply:
I kicked off a build of the first option and will test it tomorrow.

Reply:
Idk what kind of output you'll get for the (show ex), I guess it'll try to print the Error it was converted to instead of the Exception it was originally, both classes clearly derive Show so it should work.

Reply:
On the new code, I had to do:
`_err (Just (EOF s)) = do debug $ printf "%s on %s" (show (EOF s)) (show sock)` 
with just `(show s)` it was only printing the socket and skipped "end of stream received".
> <v4v: 9> on <v4v: 9>

instead of
> <v4v: 14>: end of stream reached on <v4v: 14>

Yeah, I'm not sure what:
`_err (Just ex) = do warn $ printf "%s on %s" (show ex) (show sock)`
will print since it prints the value after fromException.  Channel.hs defines `Show BounceErr` that formats the Exceptions, so we'd probably lose that...

Reply:
I dropped the `EOF` pattern and made everything go through `_err (Just ex) = warn $ printf "%s on %s" (show ex) (show sock)`, and the output was unchanged.  We should be good here.